### ISSUE TITLE
 the `upload` callback of file component is triggered repeatedly when a file is uploaded [version  >= 3.26 only]

### ISSUE THREAD

**Issue by binary-husky** [gradio-app/gradio#3853]
### Describe the bug

version 3.25 -> ok
version 3.26 -> triggered repeatedly

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

upgrade to 3.26,
register file.upload, then upload a file.

### Logs

```shell
no logs
```

### System Info

```shell
docker build
```

### Severity

annoying

**Comment by abidlabs** [comment]
Hi @binary-husky can you provide a code example we can use to reproduce this issue?

**Comment by binary-husky** [comment]
```
import os; os.environ['no_proxy'] = '*' # 避免代理网络产生意外污染
import gradio as gr

def on_file_uploaded(files):
    print('dsa')
    return

gr_L1 = lambda: gr.Row().style()
gr_L2 = lambda scale: gr.Column(scale=scale)

cancel_handles = []
with gr.Blocks(title="h", analytics_enabled=False) as demo:
    with gr_L1():
        with gr.Accordion("点击展开“文件上传区”。上传本地文件可供红色函数插件调用。", open=False) as area_file_up:
            file_upload = gr.Files(label="任何文件, 但推荐上传压缩文件(zip, tar)", file_count="multiple")
            file_upload.upload(on_file_uploaded, [file_upload], [])

demo.queue(concurrency_count=50).launch(server_name="0.0.0.0", server_port=5555)

```

**Comment by binary-husky** [comment]
> ```
> import os; os.environ['no_proxy'] = '*' # 避免代理网络产生意外污染
> import gradio as gr
> 
> def on_file_uploaded(files):
>     print('dsa')
>     return
> 
> gr_L1 = lambda: gr.Row().style()
> gr_L2 = lambda scale: gr.Column(scale=scale)
> 
> cancel_handles = []
> with gr.Blocks(title="h", analytics_enabled=False) as demo:
>     with gr_L1():
>         with gr.Accordion("点击展开“文件上传区”。上传本地文件可供红色函数插件调用。", open=False) as area_file_up:
>             file_upload = gr.Files(label="任何文件, 但推荐上传压缩文件(zip, tar)", file_count="multiple")
>             file_upload.upload(on_file_uploaded, [file_upload], [])
> 
> demo.queue(concurrency_count=50).launch(server_name="0.0.0.0", server_port=5555)
> ```

I confirm when a file is uploaded in version 3.25, `on_file_uploaded` is executed once,
but in version 3.26, `on_file_uploaded` executed once a second after a file is dragged into the file area,

**Comment by binary-husky** [comment]
my dockerfile is a regular one, nothing special or helpful:
```
FROM python:3.11

WORKDIR /target_directory
COPY requirements.txt .    
RUN pip3 install -r requirements.txt

COPY . .

CMD ["python3", "-u", "main.py"]

```

**Comment by freddyaboulton** [comment]
Was fixed by #3859 and will be released soon!

**Comment by binary-husky** [comment]
@freddyaboulton Thanks, but which future version should I target at? And where can I find documents about gradio's overall framework for developer?
